packages/ping-protect/README.md

```diff
@@ -0,0 +1,138 @@
+# Ping Protect
+
+The Ping Protect module is intended to be used along with the ForgeRock JavaScript SDK to provide the Ping Protect feature to our existing ForgeRock SDK customers.
+
+## Overall Design
+
+There are two components on the server side and two components on the client side to enable this feature. You'll need to have the following:
+
+1. ForgeRock Identity Cloud (aka ID Cloud) platform or an up-to-date ForgeRock Access Management (aka AM)
+2. PingOne tenant with Protect enabled
+3. A Ping Protect Service configured in ID Cloud or AM
+4. A journey/tree with the appropriate Protect Nodes
+5. A client application with the `@forgerock/javascript-sdk` and `@forgerock/ping-protect` installed
+
+## Quick Start for Client Application
+
+Install both modules and their latest versions:
+
+```sh
+npm install @forgerock/javascript-sdk @forgerock/ping-protect
+```
+
+The two main responsibilities of the Ping Protect feature is the initialization of the profiling and data collection and the completion and preparation of the collected data for the server. You can find the two methods on the `PIProtect` module found within the `@forgerock/ping-protect` module.
+
+- `PIProtect.init()`
+- `PIProtect.getData()`
+
+The `init` method can be called at application startup, or when you receive the `PingOneProtectInitializeCallback` callback from the server. We recommend you call `init` as soon as you can to collect as much data as possible for higher accuracy.
+
+When calling `init`, you have many different options to configure what and how the data is collected. The most important and required of these settings is the `envId`. All other settings are optional.
+
+```js
+import { PIProtect } from '@forgerock/ping-protect';
+
+// Call early in your application startup
+PIProtect.init({ envId: '02fb1243-189a-4bc7-9d6c-a919edf6447' });
+```
+
+Alternatively, you can delay the initialization until you receive the instruction from the server by way of the special callback: `PingOneProtectInitializeCallback`. To do this, you would call the `init` method when the callback is present in the journey.
+
+```js
+import { PIProtect } from '@forgerock/ping-protect';
+
+if (step.getCallbacksOfType('PingOneProtectInitializeCallback')) {
+  try {
+    // Asynchronous call
+    await PIProtect.init({ envId: '02fb1243-189a-4bc7-9d6c-a919edf6447' });
```

Comment:
Could we update this snippet to show getting the `envId` from the PingOneProtectInitializeCallback maybe? (assuming it comes through from the worker service?)